vllm/model_executor/models/chatts.py

```diff
@@ -0,0 +1,416 @@
+        patch_size = hf_config.ts['patch_size']
+        # encoded_ts_arrays: list[torch.Tensor]
+        # encoded_ts_arrays[i].shape: (num_rows, num_elements*2, num_features)
+        # num_elements*2 is used because each element is a pair of (value, mask)
+        patch_cnt = [
+            (encoded_ts_arrays[i].shape[1] // 2 + patch_size - 1) // patch_size
+            for i in range(len(encoded_ts_arrays))
+        ]
+
+        def get_replacement_qwen2_ts(item_idx: int):
+            # Use the pre-tokenized replacements
+            tokens = ts_tokens[item_idx].copy()
+            # Extend the tokens with placeholders to match the patch_cnt
+            num_placeholders = sum(1 for t in tokens if t == placeholder)
+            if num_placeholders < patch_cnt[item_idx]:
+                tokens.extend([placeholder] *
+                              (patch_cnt[item_idx] - num_placeholders))
+            # return tokens
+            return PromptUpdateDetails.select_token_id(
+                tokens,
+                embed_token_id=placeholder,
+            )
+
+        return [
+            PromptReplacement(
+                modality="timeseries",
+                target=[placeholder, placeholder + 1],
+                replacement=get_replacement_qwen2_ts,
+            )
+        ]
+
+
+@MULTIMODAL_REGISTRY.register_processor(
+    Qwen2TSMultiModalProcessor,
+    info=Qwen2TSProcessingInfo,
+    dummy_inputs=Qwen2TSDummyInputsBuilder,
+)
+class Qwen2TSForCausalLM(nn.Module, SupportsMultiModal, SupportsPP,
+                         SupportsLoRA):
+    packed_modules_mapping = {
+        "qkv_proj": [
+            "q_proj",
+            "k_proj",
+            "v_proj",
+        ],
+        "gate_up_proj": [
+            "gate_proj",
+            "up_proj",
+        ],
+    }
+
+    # To ensure correct weight loading and mapping.
+    hf_to_vllm_mapper = WeightsMapper(orig_to_new_prefix={
+        "lm_head.": "language_model.lm_head.",
+        "model.": "language_model.model.",
+    })
+
+    def __init__(self, *, vllm_config: VllmConfig, prefix: str = ""):
+        super().__init__()
+        config: PretrainedConfig = vllm_config.model_config.hf_config
+        quant_config = vllm_config.quant_config
+        multimodal_config = vllm_config.model_config.multimodal_config
+        self.config = config
+        self.multimodal_config = multimodal_config
+
+        self.ts_encoder = TimeSeriesEmbedding(config.ts)
+        self.quant_config = quant_config
+
+        self.language_model = init_vllm_registered_model(
+            vllm_config=vllm_config,
+            hf_config=config,
+            prefix=maybe_prefix(prefix, "language_model"),
+            architectures=["Qwen2ForCausalLM"],
+        )
+
+        self.make_empty_intermediate_tensors = (
+            self.language_model.make_empty_intermediate_tensors)
+
+    def _parse_and_validate_ts_input(self, **kwargs: object) -> torch.Tensor:
```

Comment:
```suggestion
    def _parse_and_validate_ts_input(self, **kwargs: object) -> Qwen2TSTimeSeriesInput:
```

I suggest wrapping the validated inputs to a `TypedDict` like `Qwen2TSTimeSeriesInput`, just like other mm models.